server/src/main/java/io/crate/role/Role.java

```diff
@@ -76,13 +92,24 @@ public static Properties of(boolean login, GenericProperties<Object> properties)
                 }
             }
             JwtProperties jwtProperties = JwtProperties.fromMap(properties.getUnsafe(JWT_KEY));
-            return new Properties(login, hash, jwtProperties);
+
+            Map<String, Object> sessionSettings = new HashMap<>();
+            for (var p : properties.stream()
+                .filter(p -> !PASSWORD_KEY.equals(p.getKey()) && !JWT_KEY.equals(p.getKey())).toList()) {
+                SessionSetting<?> sessionSetting = sessionSettingRegistry.settings().get(p.getKey());
```

Comment:
I think we need NULL check for `sessionSetting`. 

Now it's impossible because we covered all other cases (`PASSWORD_KEY` and `JWT_KEY`) but we need some protection to case when we add another non-session property to user and forget to filter it here